.github/workflows/sdk-test.yaml

```diff
@@ -0,0 +1,60 @@
+name: Speakeasy Test SDK
+
+on:
+  workflow_call:
+    inputs:
+      target:
+        description: "The specific target to test"
+        required: false
+        type: string
+      working_directory:
+        description: "The working directory for running Speakeasy CLI commands in the action"
+        required: false
+        type: string
+    secrets:
+      github_access_token:
+        description: A GitHub access token with read access to the repo
+        required: true
+      speakeasy_api_key:
+        description: The API key to use to authenticate the Speakeasy CLI
+        required: true
+jobs:
+  test:
+    name: Test SDK
+    runs-on: ubuntu-latest
+    steps:
+      - name: Tune GitHub-hosted runner network
+        uses: smorimoto/tune-github-hosted-runner-network@v1
+
+      - name: Check commit message condition
+        id: check_commit
+        run: |
+          if [[ "${{ github.event_name }}" != "push" ]]; then
+            echo "Skipping commit message check since event is not push."
+            echo "run_tests=true" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+
+          COMMIT_MESSAGE="${{ github.event.head_commit.message }}"
+          echo "Commit message: $COMMIT_MESSAGE"
+
+          # The main trigger should be PR events. This allows us to execute tests from empty commits made by our app.
+          # This is a workaround for GitHub not allowing actions to trigger from a PR made by another action.
+          if [[ "$COMMIT_MESSAGE" == *"[run-tests]"* ]]; then
```

Comment:
Slightly confused by the relationship between this `run-tests` and the `run_tests` you are writing to the output

Reply:
It's a special commit message for our empty commit flow

Reply:
Basically avoids us having duplicate runs with the PR trigger. If we make an empty commit with our app from our backend it will have [run-tests] and thus trigger tests.

Reply:
I'll try to explain better in the code comment. That was my intention.